File: test/ch21-benchmarks/bench-011-nested-loops.c
// BENCH-011: Nested Loops (1000x1000 iterations) - C
#include <stdio.h>

long long nested_loops(int outer, int inner) {
    long long total = 0;
    for (int i = 0; i < outer; i++) {
        for (int j = 0; j < inner; j++) {
            total += i * j;
        }
    }
    return total;
}

int main() {
    long long result = nested_loops(1000, 1000);
    // Expected: 249500250000
    return 0;
}

File: test/ch21-benchmarks/bench-011-nested-loops.rs
// BENCH-011: Nested Loops (1000x1000 iterations) - Rust

fn nested_loops(outer: i64, inner: i64) -> i64 {
    let mut total: i64 = 0;
    for i in 0..outer {
        for j in 0..inner {
            total += i * j;
        }
    }
    total
}

fn main() {
    let _result = nested_loops(1000, 1000);
    // Expected: 249500250000
}
